x/crosschain/migrations/v5/migrate.go

```diff
@@ -25,12 +25,60 @@ type crosschainKeeper interface {
 
 // MigrateStore migrates the x/crosschain module state from the consensus version 4 to 5
 // It resets the aborted zeta amount to use the inbound tx amount instead in situations where the outbound cctx is never created.
-func MigrateStore(
+func MigrateStore(ctx sdk.Context, crosschainKeeper crosschainKeeper, observerKeeper types.ObserverKeeper) error {
+	err := SetZetaAccounting(ctx, crosschainKeeper, observerKeeper)
+	if err != nil {
+		return err
+	}
+	ResetTestnetNonce(ctx, observerKeeper)
+
+	return nil
+}
+
+func ResetTestnetNonce(
+	ctx sdk.Context,
+	observerKeeper types.ObserverKeeper,
+) {
+	tss, found := observerKeeper.GetTSS(ctx)
+	if !found {
+		return
+	}
+	for chain, nonce := range CurrentTestnetChains() {
+		cn, found := observerKeeper.GetChainNonces(ctx, chain.ChainName.String())
+		if !found {
+			continue
```

Comment:
To confirm, Is there any chance the `GetChainNonces` returns a `found==false` but below `GetPendingNonces` returns a `found==true`, in case we are not reseting both.

Reply:
Do you mean the other way around? As in what would happen if we set chain nonces but are unable to set pending nonces for a particular chain ?  The outbounds for that chain would not get unblocked , but that is still the same as now . However the chances of that happening are almost impossible since the values exist on testnet right now . We can query to verify .
We can use tmpctx to make it atomic but I don't think its needed for this , wdyt ?